doc/contributing/contributing.md

```diff
@@ -0,0 +1,356 @@
+  and has a `slug` that identifies it.
+  An exercise consists of a `Readme` and `Code`.
+* [Readme](lib/xapi/readme.rb) - a language-independent explanation of the problem to be solved.
+* [Code](lib/xapi/code.rb) - supporting code to solve the problem. In its most basic form, it
+  consists of a test suite, but there can be other supporting files (class
+  or type definitions, metadata files, etc).
+
+There is also the concept of `Homework` which wraps the process of getting
+a user's current exercise set from exercism.io, and getting all the relevant problems.
+
+**NOTE** A _problem_ is a README and test suite/supporting code (language+slug). An _exercise_ is a problem worked by a user.
+
+### Code Arrangement
+
+The sinatra app lives in `lib/app`, and the business logic lives in
+`lib/xapi`. To determine whether something is business logic or web
+application logic, think about whether or not a different interface (say,
+perhaps a command-line interface) would also need that behavior.
+
+The tests live under `test`, and the path to the test file will mirror the
+path to the code file, except with `test` rather than `lib`:
+
+```bash
+lib/app/routes/demo.rb
+test/app/routes/demo_test.rb
+```
+
+There could be exceptions to this, but they are rare.
+
+### config.json
+
+There are a number of keys in the `config.json` of each language repository.
+
+* `problems` - these are actively served via `exercism fetch`
+* `deprecated` - these were implemented, but aren't served anymore
+* `foregone` - these must not be implemented in this language
+* `ignored` - these directories do not contain problems
+
+The `configlet` tool uses those categories to ensure that
+
+1. all the `problems` are implemented,
+2. `deprecated` problems are not actively served as problems, and
+3. `foregone` problems are not implemented.
+
+In addition, it will complain about problems that are implemented but are not
+listed in the config under the `problems` key. This is where the `ignored` key
+is useful. Ignored directories don't get flagged as unimplemented problems.
+
+A problem might be foregone for a number of reasons, typically because it's a
+bad exercise for the language.
+
+### configlet
+
+If the `config.json` file is incomplete or broken, a lot of other things break.
+To make things easier we made a small tool to help verify the config:
+https://github.com/exercism/configlet#configlet
+
+You can download the latest release from the releases page in the [configlet
+repo](https://github.com/exercism/configlet/releases), or you can use the
+`bin/fetch-configlet` command from the root of the language track repository,
+which will make a guess at what operating system and architecture you have and
+attempt to download the right one.
+
+Then, verify the config by calling `bin/configlet .` (notice the dot). This says
+_check the config of the language track that is stored right here).
```

Comment:
We're missing an underscore here (probably a typo in the original), and also there is an extra parenthesis. It should be:

_check the config of the language track that is stored right here_.